src/app/course/add-page/add-course-container/add-course-container.component.ts

```diff
@@ -12,56 +16,118 @@ import { CourseService } from '../../course.service';
 })
 export class AddCourseContainerComponent implements OnInit {
   course: Course;
+  idCourse: number;
   mode: string;
-  title: string;
+  modeTitle;
   newIndex;
   form: FormGroup;
+  date: string | Date;
 
-  constructor(public coursesService: CourseService, private router: Router,  private route: ActivatedRoute) { 
+  constructor( private store: Store<State>,
+               public coursesService: CourseService,
+               private router: Router,
+               private formBuilder: FormBuilder,
+               private route: ActivatedRoute) {
     this.route.params.pipe(
       tap(data => {
+        console.log('DATA', data); // {}
         return data;
       })
     ).subscribe(data => {
       if (data.id) {
+        this.idCourse = data.id;
         this.mode = 'Edit';
-        this.title = 'Edit course';
+        this.modeTitle = 'Edit course';
         return;
       }
       this.mode = 'Add';
-      this.title = 'Add new course'
+      this.modeTitle = 'Add new course';
     });
   }
 
   ngOnInit(): void {
     this.createForm();
+    if (this.mode === 'Edit') {
+    this.getCourseItem();
+    this.form.valueChanges.subscribe(item => {
+      this.course.name = item.title;
+      this.course.description = item.description;
+      this.course.authors = item.authors;
+      this.course.date = new Date(item.date);
+      this.course.length = Number(item.duration);
+      });
+    return;
+    }
+    // если это добавление курса
     this.createNewCourse();
   }
 
 
+  getCourseItem() {
+    this.store.dispatch(editCourse({currentId: this.idCourse}));
+    this.store.pipe(
+      select(selectCourses),
+    ).subscribe((courses: CourseState) => {
+      if (courses.currentCourse) {
+        const currentCourse = courses.currentCourse;
+        this.course = {...currentCourse};
```

Comment:
That's not clear why you store course data in two places (the `course` property and form). With this approach you have to constantly synchronize storages. It's too complicated and lead to bugs.

It would be much better to get rid of the property and store course data in the form only